Build failure on ppc32 linux since updated to indicatif-0.17.1 crate (>=maturin-0.13.3)

### Bug Description

I assume(?) this is more an issue with the crate than here, but I'm not super familiar with rust (either way felt should know), the old indicatif-0.17.0-rc.6 didn't use AtomicU64.

Don't have ppc hardware myself so haven't tried, but this is from downstream report: https://bugs.gentoo.org/874747 (full build.log available there, technically is ppc64 building with a ppc32 toolchain for testing).
```
error[E0432]: unresolved import `std::sync::atomic::AtomicU64`
 --> /var/tmp/portage/dev-util/maturin-0.13.5/work/cargo_home/gentoo/indicatif-0.17.1/src/state.rs:2:25
  |
2 | use std::sync::atomic::{AtomicU64, AtomicU8, Ordering};
  |                         ^^^^^^^^^
  |                         |
  |                         no `AtomicU64` in `sync::atomic`
  |                         help: a similar name exists in the module: `AtomicU8`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `indicatif` due to previous error
```

### Your Python version (`python -V`)

Python 3.10.6

### Your pip version (`pip -V`)

N/A

### What bindings you're using

_No response_

### Does `cargo build` work?

- [ ] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

1. normal `cargo build` with ppc32 rust toolchain in theory

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -26,7 +26,7 @@ glob = "0.3.0"
 cargo_metadata = "0.15.0"
 cargo-options = "0.3.1"
 cargo-zigbuild = "0.12.3"
-cargo-xwin = { version = "0.10.4", default-features = false }
+cargo-xwin = { version = "=0.10.2", default-features = false }
 cbindgen = { version = "0.24.2", default-features = false }
 flate2 = "1.0.18"
 goblin = "0.5.4"

--- a/Changelog.md
+++ b/Changelog.md
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * Fix `maturin develop` in Windows conda virtual environment in [#1146](https://github.com/PyO3/maturin/pull/1146)
 * Fix build for crate using `pyo3` and `build.rs` without `cdylib` crate type in [#1150](https://github.com/PyO3/maturin/pull/1150)
+* Fix build on some 32-bit platform by downgrading `indicatif` in [#1163](https://github.com/PyO3/maturin/pull/1163)
 
 ## [0.13.5] - 2022-09-27
 

--- a/deny.toml
+++ b/deny.toml
@@ -167,6 +167,7 @@ deny = [
     # is a direct dependency of the otherwise banned crate
     #{ name = "ansi_term", version = "=0.11.0", wrappers = [] },
     { name = "rustix" },
+    { name = "indicatif", version = ">=0.17.0" },
 ]
 # Certain crates/versions that will be skipped when doing duplicate detection.
 skip = [

